cmd/kubeadm/app/cmd/reset.go

```diff
@@ -48,17 +50,24 @@ func NewCmdReset(out io.Writer) *cobra.Command {
 	return cmd
 }
 
-func RunReset(out io.Writer, cmd *cobra.Command, skipPreFlight bool) error {
+type Reset struct{}
+
+func NewReset(skipPreFlight bool) (*Reset, error) {
 	if !skipPreFlight {
 		fmt.Println("Running pre-flight checks")
 		err := preflight.RunResetCheck()
 		if err != nil {
-			return &preflight.PreFlightError{Msg: err.Error()}
+			return nil, &preflight.PreFlightError{Msg: err.Error()}
```

Comment:
Ah, it's just a `struct{}`... do we really need it? We don't have dummy structs like this in neither `init.go` nor `join.go`, why is one needed here? Are you planning to add these everywhere, so that we have `init.Run()` and `join.Run()`, etc?